tools/autograd/gen_python_functions.py

```diff
@@ -1359,15 +1359,13 @@ def make_python_binding_args(declaration):
         python_binding_arguments.append(dtype_arg)
 
     if is_factory_function or is_like_or_new_function_with_options:
-        py_default_layout = '*torch::getLayout(self.options().backend())' if is_like_or_new_function_with_options else None
         layout_arg = {
-            'default': 'torch.strided',
-            'dynamic_type': 'Layout',
+            'default': 'None',
```

Comment:
You should preserve the default here; actually, the reason the old PR wasn't landable was because changing the defaults in places like this caused other parts of the codebase to break. It would be pretty cool to fix this (cc @bhosmer) but it's much bigger than a bootcamp task.

More concretely, what you need to do is rewrite `py_default_layout` to produce the C++ version (rather than the Python version) and initialize it using this code. Look at dtype to see how we do this in the other cases.